packages/enzyme/src/ShallowWrapper.js

```diff
@@ -405,22 +461,24 @@ class ShallowWrapper {
         // We don't pass the setState callback here
         // to guarantee to call the callback after finishing the render
         instance.setState(state);
+
         if (
           shouldRender &&
           !this[OPTIONS].disableLifecycleMethods &&
-          adapter.options.enableComponentDidUpdateOnSetState &&
+          lifecycles.componentDidUpdate &&
+          lifecycles.componentDidUpdate.onSetState &&
           instance
         ) {
           if (
-            adapter.options.supportGetSnapshotBeforeUpdate &&
+            lifecycles.getSnapshotBeforeUpdate &&
             typeof instance.getSnapshotBeforeUpdate === 'function'
           ) {
             const snapshot = instance.getSnapshotBeforeUpdate(prevProps, prevState);
             if (typeof instance.componentDidUpdate === 'function') {
               instance.componentDidUpdate(prevProps, prevState, snapshot);
             }
           } else if (typeof instance.componentDidUpdate === 'function') {
-            if (adapter.options.supportPrevContextArgumentOfComponentDidUpdate) {
+            if (lifecycles.componentDidUpdate.prevContext) {
```

Comment:
is `lifecycles.componentDidUpdate` guaranteed in this context as opposed to `lifecycles.componentDidUpdate.onSetState`? Otherwise we should check to see if `lifecycles.componentDidUpdate` exists.

Reply:
`lifeCycles.componentDidUpdate` is ensured by the check on line 468